sdk/containerregistry/azure-containerregistry/tests/test_container_registry_client.py

```diff
@@ -844,3 +844,31 @@ def send_in_download_blob(request: PipelineRequest, **kwargs) -> MagicMock:
                 for chunk in stream:
                     pass
             assert str(exp.value) == "The content of retrieved blob digest does not match the requested digest."
+    
+    def test_deserialize_manifest(self):
+        def get_manifest(encoding: Optional[str] = None) -> str:
+            manifest = {
+                "manifests": [
+                    {
+                        "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
+                        "imageSize": 566,
+                        "digest": "sha256:b808af65792ab617b9032c20fb12c455dc2bf5efe1af3f0ac81a129560772d35",
+                        "architecture": "unknown",
+                        "os": "unknown",
+                    }
+                ]
+            }
+            return json.dumps(manifest)
+        def send(request: PipelineRequest, **kwargs) -> MagicMock:
+            return MagicMock(
+                status_code=200,
+                content_type="application/json; charset=utf-8",
+                text=get_manifest,
+        )
+        
+        with ContainerRegistryClient(
+            endpoint=self.containerregistry_endpoint, transport = MagicMock(send=send)
+        ) as client:
+            manifests = client.list_manifest_properties(HELLO_WORLD)
+            for manifest in manifests:
+                pass
```

Comment:
This test isn't testing the enum properties in question.

Reply:
This test is to test deserializing manifest properties with value "unknown". What do you mean "testing the enum properties in question"?

Reply:
Updated